roles/v2/pool/src/lib/mining_pool/message_handler.rs

```diff
@@ -52,7 +52,7 @@ impl ParseDownstreamMiningMessages<(), NullDownstreamMiningSelector, NoRouting>
                     Err(e) => Err(e),
                 }
             })
-            .unwrap()?;
+            .map_err(|_| roles_logic_sv2::Error::PoisonLock)??;
```

Comment:
Just double checking that `??` was intended.

Reply:
yep this is another case where the function inside the safe_lock returns a result inside the safe_lock result. The double ?? handles both since they are both compatible with the return type of the function